datafusion/physical-optimizer/src/pruning.rs

```diff
@@ -1544,7 +1572,10 @@ fn build_predicate_expression(
         Ok(builder) => builder,
         // allow partial failure in predicate expression generation
         // this can still produce a useful predicate when multiple conditions are joined using AND
-        Err(_) => return unhandled_hook.handle(expr),
+        Err(e) => {
+            dbg!(format!("Error building pruning expression: {e}"));
```

Comment:
I recommend changing this to `debug!` (dbug logging rather than printing to stdout on failure)